File: BYTES.NET/WPF/MVVM/ViewModelValidationResult.vb
Namespace WPF.MVVM

    ''' <summary>
    ''' property validation result class
    ''' </summary>
    ''' <remarks></remarks>
    Public Class ViewModelValidationResult

#Region "private variable(s)"

        Private _propertyName As String = Nothing

        Private _resultLevel As ViewModelValidationRule.ResultLevels = ViewModelValidationRule.ResultLevels.IsUndefnied
        Private _outputMessage As String = ""

#End Region

#Region "public properties"

        Public ReadOnly Property PropertyName As String
            Get

                Return _propertyName

            End Get
        End Property

        Public ReadOnly Property Message As String
            Get

                Return _outputMessage

            End Get
        End Property

        Public ReadOnly Property Level As ViewModelValidationRule.ResultLevels
            Get

                Return _resultLevel

            End Get
        End Property

#End Region

#Region "public new instance method(s)"

        ''' <summary>
        ''' default new instance method
        ''' </summary>
        ''' <remarks></remarks>
        Public Sub New()
        End Sub

        ''' <summary>
        ''' overloaded new instance method
        ''' </summary>
        ''' <param name="propertyName"></param>
        ''' <param name="resultLevel"></param>
        ''' <param name="message"></param>
        Public Sub New(ByVal propertyName As String, ByVal resultLevel As ViewModelValidationRule.ResultLevels, ByVal message As String)

            'set variable(s)
            _propertyName = propertyName
            _resultLevel = resultLevel
            _outputMessage = message

        End Sub

#End Region

    End Class

End Namespace

File: src-2023/BYTES.NET.WPF/MVVM/ViewModelValidationResult.cs
//import (default) .net namespace(s) required
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace BYTES.NET.WPF.MVVM
{
    /// <summary>
    /// the generic result type for ViewModel validation
    /// </summary>
    public class ViewModelValidationResult
    {
        #region public enum(s)

        public enum ResultType
        {
            Undefined = 0,
            Information = 1,
            Warning = 2,
            Error = 3
        }

        #endregion

        #region private variable(s)

        private ViewModelValidationResult.ResultType _type = ResultType.Undefined;
        private string _message = String.Empty;

        #endregion

        #region public properties

        public ViewModelValidationResult.ResultType Type
        {
            get => _type;
        }

        public string Message
        {
            get => _message;
        }

        #endregion

        #region public new instance method(s)

        /// <summary>
        /// default new instance method
        /// </summary>
        /// <param name="message"></param>
        /// <param name="type"></param>
        public ViewModelValidationResult(string message, ResultType type = ResultType.Information)
        {
            _message = message;
            _type = type;
        }

        /// <summary>
        /// overloaded new instance method, supporting type declaration only
        /// </summary>
        /// <param name="type"></param>
        public ViewModelValidationResult(ResultType type)
        {
            _type = type;
        }

        #endregion
    }
}
